File: windows/DIWRC/SampleComponent.cs
using Microsoft.ReactNative.Managed;
using Windows.Security.ExchangeActiveSyncProvisioning;

namespace DWWRC
{
    [ReactModule]
    class SampleComponent
    {
        [ReactMethod("getDeviceModel")]
        public string GetDeviceModel()
        {
            EasClientDeviceInformation info = new EasClientDeviceInformation();
            return info.SystemProductName;
        }
    }
}

File: windows/DeviceInfo2/SampleComponent.cpp
#include "pch.h"
#include "SampleComponent.h"
#include "SampleComponent.g.cpp"
#include <../winrt/Windows.Security.ExchangeActiveSyncProvisioning.h>

//using namespace winrt::Windows::Security::ExchangeActiveSyncProvisioning;

namespace winrt::DeviceInfo2::implementation
{
    REACT_METHOD(DeviceModel, L"deviceModel");
    winrt::hstring SampleComponent::DeviceModel() noexcept
    {
        //auto info = EasClientDeviceInformation();
        return L"";//info.SystemProductName();
    }
}
